### PR TITLE
Make all top level make targets pass w/o error

### PR DESCRIPTION
**Pull request by daw3rd**
## Why are these changes needed?
Update all top level targets to pass recursive execution.
Especially change test-image in spark transforms to build the required base spark image in data-processing-lib/spark.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,11 +43,11 @@ test::
 	@# Help: Recursively $@ in all subdirs 
 	@$(MAKE) RULE=$@ .recurse
 
-publish::  
+publish::
 	@# Help: Recursively $@ in all subdirs 
 	@$(MAKE) RULE=$@ .recurse
 
-set-versions::  
+set-versions:
```
Comment (roytman) [inline Makefile:50]: @daw3rd , why do we use `::` in other rules, we don't need multiple rules for the same target, each executed independently?
and why do you remove it here? what is the difference?

File: transforms/code/code_quality/kfp_ray/Makefile [hunk 2]
```diff
@@ -19,10 +23,14 @@ test-src::
 
 test-image::
 
+publish::
+
 image::
 
 load-image::
 
+set-versions: workflow-reconcile-requirements
```
Comment (revit13) [inline transforms/code/code_quality/kfp_ray/Makefile:32]: Does it make sense to rename `workflow-reconcile-requirements` to `set-versions` to reduce the number of rules?

File: Makefile (hunk 1 above)
Comment (daw3rd) [reply Makefile:50]: The :: allows sub-makefiles to override to extend the function.  I'm not really sure we need this anymore.  However, for set-versions, I was just trying to make it consistent across all makefiles.  I guess when these are defined in the leaf makefiles, they can be whatever the leaf wants to use.  Problems seemed to arise when being inconsistent with : vs :: in makefiles that tried to override/extend one of the .make.* targets.  I'm beginning to think, the .make.* targets should all go back to : so they are thought of more as sub-routines that extendible methods.

File: transforms/code/code_quality/kfp_ray/Makefile (hunk 2 above)
Comment (daw3rd) [reply transforms/code/code_quality/kfp_ray/Makefile:32]: yes, but I didn't want to risk breaking anything that might have used the workflow target :).

**Review by roytman — APPROVED** [review]
LGTM

**Pull request merged**